app/controllers/books_controller.rb

```diff
@@ -38,21 +38,36 @@ def new
   def create
     if params[:book][:title].nil?
       isbn_params = parse_by_isbn(params[:book][:ISBN], params)
-      @book = Book.new(isbn_params)
+      if isbn_params.is_a? String
+        @book = Book.new
+        render :new
+      end
+      if Book.find_by(ISBN: isbn_params[:parameters][:ISBN]).nil?
+        file = URI.open(isbn_params[:file])
+        @book = Book.new(isbn_params[:parameters])
+        @book.cover.attach(io: file, filename: 'cover.png', content_type: 'image/png')
+      else
+        @book = Book.find_by ISBN: isbn_params[:parameters][:ISBN]
+      end
     else
       @book = Book.new(book_params)
       @book.category = Category.find(params[:book][:category_id])
     end
-    if @book.save!
-      BookOwnership.create(book: @book, user: current_user)
+    if @book.save
+      if params[:book][:wishlist] == "1"
+        Wishlist.create(book: @book, user: current_user)
```

Comment:
Should be not also create a BookOwnership? this would skip the step as it's in the else condition